### DISCUSSION TITLE
Controlling iFrame

### DISCUSSION THREAD

**Discussion by anujdivesh** [TerriaJS/terriajs#6064] [Q&A]
Hi: 
Is it possible to pass an id in the post message to TerriaJS from JavaScript.
Sample code: ```iframe.postMessage({ id:928982, initSources: mydataset}, origin);```

Thanks,

**Comment by nf-s** [comment]
Hi @anujdivesh 

`initSources` supports the [`Initialization-files`](https://github.com/TerriaJS/terriajs/blob/main/doc/customizing/initialization-files.md) spec. 

So you can add models in the `catalog` property with an `id` - which can then be used to refer to that model programmatically.

For example

```js
iframe.postMessage(
  {
    initSources: {
      catalog: [
        {
          id: "some-id",
          type: "geojson",
          name: "GeoJSON Test",
          url: "test/bike_racks.geojson",
        },
      ],
    },
  },
  origin
);
```

Does that answer your question 🙂 

Cheers,
Nick

**Reply by anujdivesh** [reply]
Hi @nf-s 
The issue is that we want to be able to return the id passed in the ```initSorces``` in the javascript popup on layer click event. An example is shown on the link:  http://phpstack-335093-2209616.cloudwaysapps.com/mbdTerria/. If we click on the pink layer, we don't have anything in the popup, while when we click on the geojson lines, we get return value in the popup. The issue is defined in detail here: https://github.com/anujdivesh/mbdTerria/blob/main/ToDo.md
